js/maps.js

```diff
@@ -179,32 +165,38 @@ function toggleLayer(event) {
 }
 
 function renderMap() {
-  const urls = Object.keys(HEATMAPS);
-  urls.forEach(function (url) {
-    HEATMAPS[url] && HEATMAPS[url].setMap(null);
-  })
-  MAP_SELECT.querySelectorAll('input:checked')
-    .forEach(function (node) {
-      const url = node.value;
-      if (HEATMAPS[url]) {
-        const heatmapData = filteredMapData(HEATMAP_DATA[url]);
-        HEATMAPS[url].setData(heatmapData);
-        HEATMAPS[url].setMap(map);
-      } else {
-        readMapFile(url);
-      }
+  return new Promise(function (resolve, reject) {
```

Comment:
If there's more than one selected checkbox, I think this will resolve for the first selected layer.